ci-operator/step-registry/osd-ccs/cluster/provision/gcp/osd-ccs-cluster-provision-gcp-commands.sh

```diff
@@ -18,25 +87,38 @@ SUBSCRIPTION_TYPE=${SUBSCRIPTION_TYPE:-"standard"}
 REGION=${REGION:-"${LEASED_RESOURCE}"}
 CLUSTER_TIMEOUT=${CLUSTER_TIMEOUT}
 
+if [ "${ENABLE_SHARED_VPC}" == "yes" ]; then
+  logger "INFO" "Extracting the Shared VPC configurations..."
+  VPC_PROJECT_ID=$(jq -r '.hostProject' "${SHARED_DIR}/xpn.json")
+  VPC_NAME=$(jq -r '.clusterNetwork' "${SHARED_DIR}/xpn.json" | grep -o '[^/]*$')
+  CONTROL_PLANE_SUBNET=$(jq -r '.controlSubnet' "${SHARED_DIR}/xpn.json" | grep -o '[^/]*$')
+  COMPUTE_SUBNET=$(jq -r '.computeSubnet' "${SHARED_DIR}/xpn.json" | grep -o '[^/]*$')
+
+  VPC_PROJECT_ID_SWITCH="--vpc-project-id ${VPC_PROJECT_ID}"
+  VPC_NAME_SWITCH="--vpc-name ${VPC_NAME}"
+  CONTROL_PLANE_SUBNET_SWITCH="--control-plane-subnet ${CONTROL_PLANE_SUBNET}"
+  COMPUTE_SUBNET_SWITCH="--compute-subnet ${COMPUTE_SUBNET}"
```

Comment:
Why not write it in one `Switch`? A switch is a set of options to support a feature.

Reply:
Per my understanding, OSD supports deploying cluster on GCP into an existing VPC, and deploying into a shared VPC is another scenario, with the former doesn't need vpc project id. Do you mean to combine all the 4 into one switch, or combine the vpc name / control-plane subnet / compute subnet into one switch?

Reply:
The switch is not corresponded to the option but to the variable defined in `if`. In the code, there is only one switch `ENABLE_SHARED_VPC` to control the four options that means if I choose Shared VPC, the four options are required.